doc/cprover-manual/goto-harness.md

```diff
@@ -309,4 +310,102 @@ main.c function function
 
 ** 0 of 1 failed (1 iterations)
 VERIFICATION SUCCESSFUL
-```
\ No newline at end of file
+```
+
+### The memory snapshot harness
+
+The `function-call` harness is used in situations in which we want the analysed
+function to work in arbitrary environment. If we want a more restricted
+environment or if we have the program in which our function will be called,
```

Comment:
or if we have an executable program in which

Reply:
Done.